src/components/CourseInfo/CourseInfo.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { Button } from '../Button/Button';
+import { timeConverter } from '../../utils/functions';
+import { useParams, Link } from 'react-router-dom';
+import { fetchDataGo, fetchDataWithId } from '../../utils/api';
+
+const CardContainer = styled.div`
+	width: 80%;
+	margin: 20px auto;
+	border: 2px solid red;
+`;
+
+const MainInfo = styled.div`
+	width: 60%;
+`;
+
+const SubInfo = styled.div`
+	width: 30%;
+`;
+
+const CourseInfo = () => {
+	const { id } = useParams();
+	const [course, setCourse] = useState();
+	const [authorsList, setAuthorsList] = useState([]);
+	const [loading, setLoading] = useState(true);
+	useEffect(() => {
+		fetchDataWithId(id, setCourse, setLoading);
```

Comment:
according to single responsibility principle, each method should perform only one function
so, `fetchDataWithId` should only perform fetch operation

Reply:
Please, set courses/loading status based on value, returned from  `fetchDataWithId` method